src/forms/signpost/SignPost.tsx

```diff
@@ -0,0 +1,156 @@
+        if (!this.state.isOpen) return;
+        const target = (evt.target as any) as HTMLElement;
+
+        const el = this.refChild.current;
+        if (!el || typeof el === "string") {
+            console.warn("wrong element type");
+            return;
+        }
+        if (!el.contains(target)) {
+            this.toggle(false);
+        }
+    };
+
+    componentDidUpdate() {
+        const {isOpen} = this.state;
+        if (!isOpen) return;
+        const childWidth = (this.refChild.current as HTMLDivElement).clientWidth;
+        const childHeight = (this.refChild.current as HTMLDivElement).clientHeight;
+        const parentWidth = (this.refParent.current as HTMLDivElement).clientWidth;
+        const parentHeight = (this.refParent.current as HTMLDivElement).clientHeight;
+
+        const {direction} = this.props;
+        let transformVal =
+            direction && calculateAxisPosition(parentHeight, parentWidth, childHeight, childWidth, direction);
+        this.setState({transformVal: transformVal});
+    }
+
+    render() {
+        const {isOpen, transformVal} = this.state;
+        const {direction, style, children, shape, size} = this.props;
+        let iconShape = shape ? shape : "dell-alert-info";
+        let iconSize = size ? size : 26;
+        let setDirection = direction ? direction : "top-left";
+        return (
+            <div ref={this.refParent} className="signpost" style={{position: "relative"}}>
+                <Button
+                    className={classNames([
+                        "signpost-action", //prettier
+                        "btn-small",
+                        "btn-link",
+                        "signpost-trigger",
+                        isOpen && "active",
+                    ])}
+                    onClick={() => this.handleButtonClick()}
+                >
+                    <Icon shape={iconShape} size={iconSize} />
+                </Button>
+                {isOpen && (
+                    <div
+                        ref={this.refChild}
+                        className={classNames([
+                            "signpost-content",
+                            setDirection, //prettier
+                        ])}
+                        style={{
+                            ...style,
+                            position: "absolute",
+                            top: "0px",
+                            bottom: "auto",
+                            left: "0px",
+                            right: "auto",
+                            transform: transformVal,
+                        }}
+                    >
+                        <div className="signpost-flex-wrap">
+                            <div className="popover-pointer" />
+                            <div className="signpost-content-header">
+                                <Button className="signpost-action close" onClick={() => this.handleButtonClick()}>
```

Comment:
Sorry, probably I was not correct in the explanation. I meant that if you have a pattern for defining function via arrow function you would use onClick={this.handleButtonClick}. Where handleButtonClick = () => {......}

You can read some ideas here https://github.com/facebook/react/issues/9851

But if you choose use bind. the previous code is correct, but better the bind call move to constructor() like in the link above

Reply:
Sorry I misunderstood, corrected the function.
Thank you.